common/artifact-manager/src/main/java/com/vmware/pscoe/iac/artifact/rest/client/vrli/AbstractRestClientVrli.java

```diff
@@ -33,114 +33,120 @@
-			}
-			throw new RuntimeException(String.format("Error importing content pack '%s' to VRLI: %s", contentPackName, e.getMessage()));
-		} catch (RestClientException e) {
-			throw new RuntimeException(String.format("REST client error during import of content pack '%s' to VRLI: %s", contentPackName, e.getMessage()));
-		}
-	}
-	protected RestClientVrops getVropsRestClient() {
-		if (this.vropsRestClient != null) {
-			return this.vropsRestClient;
-		}
-
-		ConfigurationVrops vropsConfiguration;
-		Properties vropsConfigProperties = new Properties();
-		vropsConfigProperties.put(Configuration.HOST, configuration.getIntegrationVropsAuthHost());
-		vropsConfigProperties.put(Configuration.PORT, configuration.getIntegrationVropsAuthPort());
-		vropsConfigProperties.put(Configuration.USERNAME, configuration.getIntegrationVropsAuthUser());
-		vropsConfigProperties.put(Configuration.PASSWORD, configuration.getIntegrationVropsAuthPassword());
-		vropsConfigProperties.put(ConfigurationVrli.INTEGRATION_VROPS_AUTH_SOURCE, configuration.getIntegrationVropsAuthSource());
-		vropsConfigProperties.put(ConfigurationVrops.VROPS_DASHBOARD_USER, "admin");
-		vropsConfigProperties.put(ConfigurationVrops.VROPS_REST_USER, configuration.getIntegrationVropsAuthUser());
-		vropsConfigProperties.put(ConfigurationVrops.VROPS_REST_PASSWORD, configuration.getIntegrationVropsAuthPassword());
-		vropsConfigProperties.put(ConfigurationVrops.VROPS_REST_AUTH_SOURCE, configuration.getIntegrationVropsAuthSource());
-		vropsConfigProperties.put(ConfigurationVrops.SSH_PORT, "22");
-
-		try {
-			vropsConfiguration = ConfigurationVrops.fromProperties(vropsConfigProperties);
-		} catch (ConfigurationException e) {
-			throw new RuntimeException(
-				String.format("Unable to update vCOPs integration for alert, vROPs integration configuration failed with message: %s", e.getMessage()));
-		}
-
-		return new RestClientVrops(vropsConfiguration, restTemplate);
-	}
+    protected String apiPrefix;
+    protected RestTemplate restTemplate;
+    protected ConfigurationVrli configuration;
+    protected String vrliVersion;
+    protected RestClientVrops vropsRestClient;
+    protected Logger logger;
+    protected static final String CONTENT_PACKS_API = "/content/contentpack";
+    protected static final String CONTENT_PACKS_LIST_API = "/content/contentpack/list";
+    protected static final String ALERTS_API = "/alerts";
+    protected static final String VRLI_RESOURCE_KEY_TYPE = "LogInsightLogServer";
+    private static final String OVERWRITE_MODE = "OVERWRITE";
+
+    protected AbstractRestClientVrli(String apiPrefix, ConfigurationVrli configuration, RestTemplate restTemplate) {
+        this.apiPrefix = apiPrefix;
+        this.restTemplate = restTemplate;
+        this.configuration = configuration;
+        this.vropsRestClient = getVropsRestClient();
```

Comment:
Why is this needed?

Reply:
For rewriting the data for the vCOPs enabled alerts (recipients->vrops->vcopsResourceKindKey)